crates/primitives/src/transaction/mod.rs

```diff
@@ -620,6 +620,59 @@ impl TransactionSigned {
         initial_tx.hash = initial_tx.recalculate_hash();
         initial_tx
     }
+
+    pub fn decode_enveloped(&self, tx: Bytes) -> Result<Self, DecodeError> {
+        let mut data = tx.as_ref();
+
+        let transaction = if data[0] > 0x7f {
+            // legacy transaction
+            todo!()
```

Comment:
here we need to decode the legacy transaction from rlp.

I'd like to move:

https://github.com/paradigmxyz/reth/blob/314ea0883b1062375ad86577be1fccc39233739d/crates/primitives/src/transaction/mod.rs#L694-L710

to a standalone function

Reply:
alright, I thought the impl was going to be different from the ```decode``` impl because the input was ```[id, rlp(typedtx)]``` ?

Reply:
only for types transactions, for legacy it is the same.